src/components/MagicLinkForm.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router';
+
+import { RecaptchaAction } from '@graasp/sdk';
+
+import { LoadingButton } from '@mui/lab';
+
+import { useAuthTranslation } from '../config/i18n';
+import { SIGN_IN_MAGIC_LINK_SUCCESS_PATH } from '../config/paths';
+import { mutations } from '../config/queryClient';
+import { EMAIL_SIGN_IN_FIELD_ID, SIGN_IN_BUTTON_ID } from '../config/selectors';
+import { useRecaptcha } from '../context/RecaptchaContext';
+import { useMobileAppLogin } from '../hooks/mobile';
+import { useRedirection } from '../hooks/searchParams';
+import { AUTH } from '../langs/constants';
+import { emailValidator } from '../utils/validation';
+import EmailInput from './EmailInput';
+import ErrorDisplay from './common/ErrorDisplay';
+
+const { SIGN_IN_BUTTON } = AUTH;
+
+const MagicLinkForm = () => {
+  const navigate = useNavigate();
+  const { t } = useAuthTranslation();
+  const { executeCaptcha } = useRecaptcha();
+
+  const { isMobile, challenge } = useMobileAppLogin();
+  const redirect = useRedirection();
+
+  const [email, setEmail] = useState('');
+
+  // enable validation after first click
+  const [shouldValidate, setShouldValidate] = useState(false);
+
+  const {
+    mutateAsync: signIn,
+    isLoading: isLoadingSignIn,
+    error: webSignInError,
+  } = mutations.useSignIn();
+  const {
+    mutateAsync: mobileSignIn,
+    isLoading: isLoadingMobileSignIn,
+    error: mobileSignInError,
+  } = mutations.useMobileSignIn();
```

Comment:
it is more a general question here but isn't it possible to let the query client handle the choice to send to the mobile web routes ? This could simplify this, because it looks like duplicated code. Also I think we will be able to merge them in the backend ?